webui/react/src/components/CreateUserModal.tsx

```diff
@@ -177,7 +177,6 @@ const CreateUserModalComponent: React.FC<Props> = ({
       data-test-component="createUserModal"
       size="small"
       submit={{
-        disabled: hasErrors(form),
```

Comment:
i think we wanna keep `disabled` for usability. probably we can use `useWatch`

Reply:
No idea how to do this but I'll give it a shot.

Reply:
I couldn't figure out a way to do it with `Form.useWatch` -- the `disabled` field only takes a boolean and can't be made to accept a lambda or "hook". Same with the `submit` field only taking a flat object, and the top-level function really wanting to be the only place where the `Modal`'s fields are defined.

Also, `useWatch` seems to operate on form fields, which is handy for other cases, but I couldn't figure out a way to make it return computed values like `hasErrors(form)` or the `form.getFieldsError()` method it wraps.

As kind of a workaround, I ended up using `useState` instead, and making the form responsible for toggling the `canSubmit` state, which is then used to determine whether the `Save` button is disabled.

I'd be interested in learning more about component lifecycles if we do a webui 101 :)

Reply:
The current approach doesn't work in `Edit User` model (this modal is shared for `add user` and `edit user` unfortunately)

https://github.com/determined-ai/determined/assets/109113616/912493c6-fcbb-4228-9fac-36e961ac93dd

probably this example might help?

https://ant.design/components/form#components-form-demo-validate-trigger